pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -93,17 +93,11 @@ func (configgen *ConfigGeneratorImpl) BuildClusters(env *model.Environment, prox
 
 	outboundClusters := configgen.buildOutboundClusters(env, proxy, push)
 
-	if env.Mesh.LocalityLbSetting != nil {
-		// apply load balancer setting fot cluster endpoints
-		applyLocalityLBSetting(proxy.Locality, outboundClusters, env.Mesh.LocalityLbSetting)
-	}
-	// Add a blackhole and passthrough cluster for catching traffic to unresolved routes
-	// DO NOT CALL PLUGINS for these two clusters.
-	outboundClusters = append(outboundClusters, buildBlackHoleCluster(env), buildDefaultPassthroughCluster(env, proxy))
-
 	switch proxy.Type {
 	case model.SidecarProxy:
 		outboundClusters = envoyfilter.ApplyClusterPatches(networking.EnvoyFilter_SIDECAR_OUTBOUND, proxy, push, outboundClusters)
+		// apply load balancer setting fot cluster endpoints
+		applyLocalityLBSetting(proxy.Locality, outboundClusters, env.Mesh.LocalityLbSetting)
```

Comment:
The `env.Mesh.LocalityLbSetting` is in the `applyLocalityLBSetting`.

Reply:
Should we do that patch before or after applying locality? Seems like after?

Reply:
yeah. it should be after

Reply:
make sense

Reply:
always apply patch at last